app/src/main/java/tech/redroma/yoching/activities/ReadActivity.kt

```diff
@@ -103,35 +135,55 @@ class ReadActivity : AppCompatActivity()
         lateinit var whatsUpTitle: TextView
         lateinit var whatsUpBody: DocumentView
 
-        override fun inflate(activity: AppCompatActivity)
+        fun inflate()
+        {
+            //Pull out the Views
+            actionBarToolbar = findView(id.action_toolbar)
+            actionBarTitle = findView(id.yo_action_bar_title)
+            wrexagramImage = findView(id.wrexagram_image)
+            wrexagramTitle = findView(id.wrexagram_title)
+            duplicateTitle = findView(R.id.wrexagram_title_duplicate)
+            body = findView(id.wrexagram_body)
+            whatsUpTitle = findView(R.id.whats_up_title)
+            whatsUpBody = findView(R.id.whats_up_body)
+
+            setFonts()
+            setupBackButton()
+
+            body.layoutParams.height = WRAP_CONTENT
+            whatsUpBody.layoutParams.height = WRAP_CONTENT
+            wrexagramImage.setImageDrawable(null)
+        }
+
+        private fun setFonts()
         {
-            activity.perform {
-
-                //Pull out the Views
-                actionBarToolbar = findView(id.action_toolbar)
-                actionBarTitle = findView(id.yo_action_bar_title)
-                wrexagramImage = findView(id.wrexagram_image)
-                wrexagramTitle = findView(id.wrexagram_title)
-                duplicateTitle = findView(R.id.wrexagram_title_duplicate)
-                body = findView(id.wrexagram_body)
-                whatsUpTitle = findView(R.id.whats_up_title)
-                whatsUpBody = findView(R.id.whats_up_body)
-
-                //Action bar code
-                setSupportActionBar(actionBarToolbar)
-                supportActionBar?.setHomeButtonEnabled(true)
-                supportActionBar?.setDisplayHomeAsUpEnabled(true)
-                title = ""
-                val backButton = resources.getDrawable(R.drawable.arrow_back)
-                supportActionBar?.setHomeAsUpIndicator(backButton)
-
-                //Set the typefaces
-                whatsUpTitle.typeface = exoExtraBold()
-                actionBarTitle.typeface = exoBlack()
-                wrexagramTitle.typeface = exoBlack()
-
-                body.layoutParams.height = WRAP_CONTENT
-                whatsUpBody.layoutParams.height = WRAP_CONTENT
+            //Set the typefaces
```

Comment:
Put this into a separate function![bitmoji](https://render.bitstrips.com/v2/cpanel/8736904-288439948_4-s1-v1.png?transparent=1&palette=1&width=246)